x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -575,6 +625,25 @@ void incrementDifferentiatorAndCreateNewProfile(Subject subject, ProfileDocument
             listener.onFailure(new ElasticsearchException("profile uid [{}] does not contain a differentiator", uid));
             return;
         }
+
+        final DomainConfig domainConfig = getDomainConfigForSubject(subject);
+        if (domainConfig != null && differentiatorString.equals(domainConfig.suffix())) {
```

Comment:
I think this should be:
`if (domainConfig != null && domainConfig.suffix() != null)`

This is the case where a user that is part of a domain that doesn't do the auto-increment logic, tries to activate an existing profile of another user, which is an error (irrespective of the suffix value).

Reply:
In practice, the `differentiatorString` should always equal to `domainConfig.suffix()` because when the code reaches here, it is because a profile with UID ending with `domainConfig.suffix()` is owned by the user.

But I think you are also correct in that the logic should be strict and independant here. I'll update as suggested.

Reply:
Yeah, it should be an internal error if they're not equal.